### ISSUE TITLE
Add Product Type Unit Tests

### ISSUE THREAD

**Issue by devGregA** [DefectDojo/django-DefectDojo#1131]
Please use the following example at  as a template:
https://github.com/DefectDojo/django-DefectDojo/blob/dev/tests/Product_unit_test_example.py

Implement the following tests:
- Add
- Edit
 
Master List and Overview: https://docs.google.com/document/d/1dSS7tWuQukAX3VWtzPAlOfFNoqLQh-mBHBfmsu2WlU0/edit?usp=sharing

**Comment by ppiotr3k** [comment]
@devGregA, have the use of [django-webtest](https://pypi.org/project/django-webtest/) been considerated/evaluated?

For smoke/integration tests, it is a very light (i.e. fast to execute) approach for testing models and views, and being the integration for Django of the awesome [Pylons WebTest](https://docs.pylonsproject.org/projects/webtest/en/latest/), best in class for testing WSGI applications, it goes straight to the point (i.e. very fast to write tests).

For example, opening the index page, clicking on a button and checking the resulting page contains a given string is a 2 lines job:
```python
    def test_products_link(self):
        # Load the index, pretending to be logged in as user `devGregA`
        index = self.app.get('/', user='devGregA')

        # Click on the on the '<a href="/products/">Products</a>' link, check that it works
        # (no exception is raised and code 200) that the result page contains 'My Products'
        assert 'My Products' in index.click('Products')
```

Same for submitting a form, very simple, as shown in the example from [django-webtest](https://pypi.org/project/django-webtest/) page:
```python
class AuthTest(WebTest):
    fixtures = ['users.json']

    def test_login(self)
        form = self.app.get(reverse('auth_login')).form
        form['username'] = 'foo'
        form['password'] = 'bar'
        response = form.submit().follow()
        self.assertEqual(response.context['user'].username, 'foo')
```
Generally speaking, all form-related operations (checkbox, radio, ...) are very simple to use ([WebTest doc](https://docs.pylonsproject.org/projects/webtest/en/latest/forms.html)).


Last point, [django-webtest](https://pypi.org/project/django-webtest/) integrates perfectly with [pytest-django](https://pytest-django.readthedocs.io/en/latest/), which is the integration for Django of [pytest](https://docs.pytest.org/en/latest/contents.html). The _[Why would I use this instead of Django’s manage.py test command?](https://pytest-django.readthedocs.io/en/latest/#why-would-i-use-this-instead-of-django-s-manage-py-test-command)_ section explains it pretty well.

Besides the reduced boilerplate and increased test runner speed, [fixtures](https://docs.pytest.org/en/latest/fixture.html) and [parametrization](https://docs.pytest.org/en/latest/parametrize.html) are for me the biggest game-changers. Fixtures for example can provide you whatever external resource (SMTP server, HTTP server, LDAP server, ...) is needed, but without hitting the actual resource. As for parametrization, it allows for example to rerun the same test several times but with different input values, allowing a huge factorization of the test suite code.

As I've been using WebTest for 10+ years now, with pytest for 6+ years, I can hardly go back to the vanilla Selenium + unittest approach :wink:.
I still keep using Selenium with a headless chromedriver, through [pytest-selenium](https://pytest-selenium.readthedocs.io/en/latest/index.html), for automating screenshots when visual validation is required.

**Comment by dr3dd589** [comment]
Start working on it.

**Comment by Maffooch** [comment]
Merged under PR#1153
